src/screens/DiscoveryScreenV2/fragment/UsersFragment.js

```diff
@@ -38,22 +37,15 @@ const UsersFragment = ({
   searchText,
   isUser
 }) => {
-  const [discovery, discoveryDispatch] = React.useContext(Context).discovery;
   const [profile] = React.useContext(Context).profile;
   const navigation = useNavigation();
   const [client] = React.useContext(Context).client;
+  const {exhangeFollower, users, updateFollowDiscoveryContext} = useDiscovery();
 
   const route = useRoute();
 
   const [myId, setMyId] = React.useState('');
 
```

Comment:
<img src="https://avatars.githubusercontent.com/in/347564?s=41" alt="Image description" width="20" height="20">   CodeRabbit

The use of `useDiscovery` hook is a good change as it encapsulates the logic related to discovery and makes the component cleaner. However, ensure that the `exhangeFollower`, `users`, and `updateFollowDiscoveryContext` functions returned by `useDiscovery` are implemented correctly and tested thoroughly.



<!-- This is an auto-generated comment by OSS CodeRabbit -->